src/tools/collectors.rs

```diff
@@ -0,0 +1,144 @@
+use std::collections::BTreeSet;
+
+use super::prelude::*;
+
+/// Collect data about blob
+#[derive(Debug)]
+pub struct BlobSummaryCollector {
+    header: BlobHeader,
+    records: usize,
+    keys: BTreeSet<Vec<u8>>,
+}
+
+impl BlobSummaryCollector {
+    /// Collect data about blob by path
+    pub fn from_path(path: &Path) -> Result<Self> {
```

Comment:
We should be able to collect both short and full information about BLOB. Short information is just an information from BLOB header, full is information obtained by BLOB scanning. I suggest to store a path to the file inside struct and add 2 functions: first loads data from header, second - loads data by scanning the whole BLOB file. `from_path` should receive a `bool` argument, that allows full loading when it is true.